src/tscore/Regex.cc

```diff
@@ -238,7 +234,7 @@ DFA::match(const char *str, int length) const
   dfa_pattern *p = _my_patterns;
 
   while (p) {
-    rc = p->_re->exec(str, length);
+    rc = p->_re->exec({str, size_t(length)});
```

Comment:
Maybe a future enchantment is to change the prototypes to use size_t instead of int for things as length as well ?

Reply:
Step by step, bro. The next PR will convert DFA to be `string_view` supporting, at which point this ugliness moves "up a level". I.e. `DFA::match` will take `string_view` that can then be passed to `Regex`. But then the code sites that call `DFA::match` will need this. Until we convert everything, there will be an interface where this kind of thing has to be done, so we'll just have to live with it for now. Having pondered the issue quite extensively, I concluded the bottom up approach was the best way to move forward.